pkg/operator/config.go

```diff
@@ -2,19 +2,37 @@ package operator
 
 import (
 	"encoding/json"
+	"time"
 )
 
 type unsupportedServiceCAConfig struct {
+	CAConfig caConfig `json:"caConfig"`
+
 	TimeBasedRotation timeBasedRotationConfig `json:"timeBasedRotation"`
 
 	ForceRotation forceRotationConfig `json:"forceRotation"`
 }
 
+type caConfig struct {
+	// validityDuration determines how long a new signing CA will be valid for from
+	// the time that it is generated. If unspecified or 0, the CA will be valid for 1
+	// year.
+	// +optional
+	ValidityDuration time.Duration `json:"validityDuration"`
+}
+
 type timeBasedRotationConfig struct {
+	// enabled determines whether automatic rotation will occur when the signing CA
+	// has less than a minimum validity duration.
+	// +optional
 	Enabled bool `json:"enabled"`
 }
 
 type forceRotationConfig struct {
+	// reason indicates why a rotation of the signing CA should be forced. If the
+	// reason is not empty and has not been recorded as an annotation on the signing
+	// secret, the rotation of the signing CA will be triggered at most once.
+	// +optional
 	Reason string `json:"reason"`
```

Comment:
API lgtm

Reply:
So you do not share @stlaz's concerns regarding `ValidityDuration` being specified in at most hours?

Reply:
Probably >= 1h makes sense. So either document the minimum or switch to hours as integer.

Reply:
What would be the value in switching to an integer representing hours given that `metav1.Duration` supports providing a time in hours (e.g. `1h`) and supporting a duration in minutes (e.g. `20m`) is potentially useful for testing?

Reply:
See @liggitt's talk at contributor summit: `metav1.Duration` is considered a failure.

Reply:
I don't see the video up yet, but I'm assuming you're suggesting that metav1.Duration not be used? That reinforces my suggestion for `ValidityDurationForTesting` and denominating it in minutes.  If `ValidityDuration` becomes a thing to set the CA expiry for production use, it would likely be in days.

Reply:
ValidityDurationForTesting is fine with me as well. Let's decide and move forward.

Reply:
I've update the name to `ValidityDurationForTesting` and left it as `metav1.Duration`, hopefully that makes sense for you.